### PR TITLE
MultiLanguage & MultiDomain on the same server

### PR DESCRIPTION
**Pull request by zzuutt**
Suite a la PR #2381 
This PR, translate the url into the requested language

example:
langue : FR  Domain: http;//www.domaineFR.com   page product: produit1
langue : EN Domain: http://www.domaineEN.com   page product: product_1

on the page : http://www.domaineFR.com/produit1
this request
```{url path="{navigate to="current"}" lang='en_US'}```
return 
``` http://www.domaineEN.com/product_1```

### CODE REVIEW

File: local/modules/TheliaSmarty/Template/Plugins/UrlGenerator.php [hunk 1]
```diff
@@ -119,6 +122,36 @@ public function generateUrlFunction($params, &$smarty)
...
+            $view = $request->attributes->get('_view', null);
+            $viewId = $view === null ? null : $request->query->get($view . '_id', null);
+
+            if (null !== $requestedLangCodeOrLocale) {
+                if (strlen($requestedLangCodeOrLocale) > 2) {
+                    $lang = LangQuery::create()->findOneByLocale($requestedLangCodeOrLocale);
+                } else {
+                    $lang = LangQuery::create()->findOneByCode($requestedLangCodeOrLocale);
+                }
+                $urlRewrite = RewritingUrlQuery::create()
```
Comment (roadster31) [inline local/modules/TheliaSmarty/Template/Plugins/UrlGenerator.php]: You can move this request inside the ` if (ConfigQuery::isMultiDomainActivated()) {` test to save on request

File: local/modules/TheliaSmarty/Template/Plugins/UrlGenerator.php [hunk 2]
```diff
@@ -119,6 +122,36 @@ public function generateUrlFunction($params, &$smarty)
...
+                    ->filterByView($view)
+                    ->filterByViewId($viewId)
+                    ->filterByViewLocale($lang->getLocale())
+                    ->findOneByRedirected(null)
+                ;
+
```
Comment (roadster31) [inline local/modules/TheliaSmarty/Template/Plugins/UrlGenerator.php]: You can move this request inside the ` if (ConfigQuery::isMultiDomainActivated()) {` test below to save one request

File: local/modules/TheliaSmarty/Template/Plugins/UrlGenerator.php [hunk 3]
```diff
@@ -119,6 +122,36 @@ public function generateUrlFunction($params, &$smarty)
...
+                if (ConfigQuery::isMultiDomainActivated()) {
+                    $domainUrl = $lang->getUrl();
+                    $path = '';
+                    if(null != $urlRewrite){
+                        $path = "/".$urlRewrite->getUrl();
```
Comment (roadster31) [inline local/modules/TheliaSmarty/Template/Plugins/UrlGenerator.php]: Why do you add the front '/' ?

File: local/modules/TheliaSmarty/Template/Plugins/UrlGenerator.php [hunk 4]
```diff
@@ -119,6 +122,36 @@ public function generateUrlFunction($params, &$smarty)
...
+                    ->filterByView($view)
+                    ->filterByViewId($viewId)
+                    ->filterByViewLocale($lang->getLocale())
+                    ->findOneByRedirected(null)
+                ;
+
+                if (ConfigQuery::isMultiDomainActivated()) {
+                    $domainUrl = $lang->getUrl();
+                    $path = '';
+                    if(null != $urlRewrite){
```
Comment (roadster31) [inline local/modules/TheliaSmarty/Template/Plugins/UrlGenerator.php]: One space after `if`, please

File: local/modules/TheliaSmarty/Template/Plugins/UrlGenerator.php [hunk 5]
```diff
@@ -119,6 +122,37 @@ public function generateUrlFunction($params, &$smarty)
...
+                    ;
+                    
+                    $domainUrl = $lang->getUrl();
+                    $path = '';
+                    if (null != $urlRewrite) {
+                        $path = "/".$urlRewrite->getUrl();
+                    }
+                    $baseUrl = $request->getBaseUrl();
+                    $url = rtrim($domainUrl, "/").$baseUrl.$path;
+                }
```
Comment (roadster31) [inline local/modules/TheliaSmarty/Template/Plugins/UrlGenerator.php]: You can write something like  `$url = rtrim($lang->getUrl(), "/").$request->getBaseUrl().$path;` and save two useless variable affectations

(promise, after this change I'll merge the PR :) )

**Comment by zzuutt** [comment]
de rien c’est avec plaisir ;)

**Pull request merged**